polytracker/database.py

```diff
@@ -875,31 +885,40 @@ class DBTaintOutput(Base, TaintOutput):  # type: ignore
     __tablename__ = "output_taint"
     input_id = Column(Integer, ForeignKey("input.id"))
     offset = Column(BigInteger)
-    label = Column(BigInteger)
+    label = Column(BigInteger, ForeignKey("taint_forest.label"))
+    taint_forest_node: "DBTaintForestNode" = relationship("DBTaintForestNode", foreign_keys=[label],
```

Comment:
Why quoted "DBTaintForestNode"?

Reply:
It's one of the quirks of Python type-hinting. We don't define the `DBTaintForestNode` class until later in the file, so we can't reference the type yet. Referencing the type explicitly here would cause an error immediately when the module is loaded. The workaround is to sort-of forward declare the type by putting it in a string.

Reply:
Ahh... Of course. It makes sense. I wasn't aware of this.